### PR TITLE
Solve quick_sort probem

### LINKED ISSUE #15: Challenge: Quick_sort
Solve Quick_sort

write quick_sort realization
add documentation
all tests are green

-->

### PR DESCRIPTION
**Pull request by YaninaChukanova**
- [ ] PR's title includes the challenge's name and language
- [x] PR has short and clear description of the challenge
- [x] PR has appropriate labels and milestones for easy identification
- [x] PR it is assigned to the owner
- [x] reviewers are assigned
- [x] the PR contributes only one focused change
- [x] the branch is up to date with `main`/`master`
- [x] the code works when pulled and run locally
- [x] all conflicts are resolved (if any)
- [ ] It is linked to an issue in the appropriate column of the project board
- [ ] feedback is addressed (if any, and if it is appropriate feedback.)

## README Documentation

- [x] The solution is documented in `/src/README.md`
- [x] The markdown source is formatted
- [x] Spelling and grammar is correct in all text
- [x] The markdown looks correct when you preview the file
- [ ] All links and images work
- [ ] The README documents the solution's behavior, strategy(ies) and implementation(s)
- [ ] There are use cases in code blocks to illustrate the function's behavior

## Python Files

- [x] There is a module header
- [x] There is a module docstring
- [x] File names are in snake_case
- [x] Test files are named `test_<module_name>.py`

## Function Docstring

- [x] Behavior description
- [x] Parameter description
- [x] Return value description
- [x] Include any assumptions
- [x] Include 3 or more (passing!) doctests
- [ ] Include 1-2 use cases (if necessary)

## Function Implementation

- [x] The solution is not is not copied from [djeada](https://github.com/djeada/Algorithms-And-Data-Structures)
- [x] The code is auto-formatted
- [x] The code has no (reasonable) linting mistakes
- [x] Variables are named with snake_case
- [x] The function has a clear and helpful name
- [x] The file's name matches the function name
- [x] The code follows the strategy as simply as possible
- [x] Variable names are clear and helpful
- [x] Comments explain the strategy (if necessary)
- [ ] There are type annotations
- [ ] (challenge) The code includes defensive assertions

## Unit Test Suite

- [x] The test class has a helpful name in PascalCase
- [x] The test class has a docstring
- Each unit test has
  - [x] A helpful name
  - [x] A clear docstring
  - [x] Only one assertion
- [x] All tests pass
- [x] There are tests for defensive assertions
- [x] There are tests for boundary cases
- [ ] Test suite includes black-box and glass-box tests

### CODE REVIEW

File: src/sorting/python/quick_sort/README.md
Comment (AnnaShliapkina) [inline src/sorting/python/quick_sort/README.md:1]: All explanation you added in README you can (and should) put in docstrings before each function. This file also need documentation.

README filled in a little bit different way it should be. Check requirements for README.md

File: src/sorting/python/quick_sort/src/quick_sort.py
Comment (AnnaShliapkina) [inline src/sorting/python/quick_sort/src/quick_sort.py:1]: Add behavior, explanation of attributes and returns.

**Review by AnnaShliapkina — CHANGES_REQUESTED** [review]

**Review by DmytroLyt — CHANGES_REQUESTED** [review]
-  PR's title includes the challenge's name and language (language doesn`t exist)
- PR it is assigned to the owner ( PR is not assigned at all, Need to assign)
- The README documents the solution's behavior, strategy(ies) and implementation(s) ( Need to be structured and not clean explanation)
- There is a module header (Doesn`t have, Need to add)
-  There is a module docstring (Doesn`t have, Need to add)
- The solution is not is not copied from [djeada](https://github.com/djeada/Algorithms-And-Data-Structures) _( Code is copied, just was changed variables names, need another solution)_
- The test class has a docstring  (Doesn`t have, Need to add)

**Review by DmytroLyt — APPROVED** [review]
Issues solved

File: src/sorting/python/quick_sort/README.md
Comment (AnnaShliapkina) [reply src/sorting/python/quick_sort/README.md:1]: Checked. Everything solved

**Review by AnnaShliapkina — APPROVED** [review]

**Pull request merged**